organization/src/integrationTest/kotlin/com/cosmotech/organization/service/OrganizationServiceIntegrationTest.kt

```diff
@@ -1283,1447 +1283,1502 @@ class OrganizationServiceIntegrationTest : CsmRedisTestBase() {
+    } else {
+      assertNotNull(organizationRegistered)
     }
+  }
 
-    /** Create default test Connector */
-    internal fun createTestOrganization(name: String): Organization {
-      return Organization(name = name)
-    }
+  /** Run a test with current user as Organization.User */
+  private fun runAsOrganizationUser() {
+    mockkStatic(::getCurrentAuthentication)
+    every { getCurrentAuthentication() } returns mockk<BearerTokenAuthentication>()
+    every { getCurrentAccountIdentifier(any()) } returns TEST_USER_ID
+    every { getCurrentAuthenticatedUserName(csmPlatformProperties) } returns "test.user"
+    every { getCurrentAuthenticatedRoles(any()) } returns listOf(ROLE_ORGANIZATION_USER)
+  }
+
+  /** Run a test with current user as Platform.Admin */
+  private fun runAsPlatformAdmin() {
+    mockkStatic(::getCurrentAuthentication)
+    every { getCurrentAuthentication() } returns mockk<BearerTokenAuthentication>()
+    every { getCurrentAccountIdentifier(any()) } returns TEST_ADMIN_USER_ID
+    every { getCurrentAuthenticatedUserName(csmPlatformProperties) } returns "test.admin"
+    every { getCurrentAuthenticatedRoles(any()) } returns listOf(ROLE_PLATFORM_ADMIN)
+  }
 
-    /** Create default test Connector */
-    internal fun createTestOrganizationWithSimpleSecurity(
-        name: String,
-        userName: String,
-        defaultSecurity: String,
-        role: String,
-    ): Organization {
-      return Organization(
-          id = "organization_id",
-          name = name,
-          security =
-              OrganizationSecurity(
-                  default = defaultSecurity,
-                  accessControlList = mutableListOf(OrganizationAccessControl(userName, role))))
+  /** Run a test with different Organization.User */
+  private fun runAsDifferentOrganizationUser() {
+    every { getCurrentAccountIdentifier(any()) } returns OTHER_TEST_USER_ID
+    every { getCurrentAuthenticatedUserName(csmPlatformProperties) } returns "test.other.user"
+    every { getCurrentAuthenticatedRoles(any()) } returns listOf(ROLE_ORGANIZATION_USER)
+  }
+
+  /** Organization batch creation */
+  internal fun batchOrganizationCreation(numberOfOrganizationToCreate: Int) {
+    logger.info("Creating $numberOfOrganizationToCreate connectors...")
+    IntRange(1, numberOfOrganizationToCreate).forEach {
+      val newOrganization = createTestOrganization("o-connector-test-$it")
+      organizationApiService.registerOrganization(newOrganization)
     }
+  }
 
-    internal fun testFindAllOrganizations(page: Int?, size: Int?, expectedResultSize: Int) {
-      val organizationList = organizationApiService.findAllOrganizations(page, size)
-      logger.info("Organization list retrieved contains : ${organizationList.size} elements")
-      assertEquals(expectedResultSize, organizationList.size)
+  /** Organization batch creation with RBAC */
+  internal fun batchOrganizationCreationWithRBAC(userId: String): Int {
```

Comment:
naming: createOrganizationsWithAllCombinationOfRole

Reply:
add comment to explain

Reply:
I changed the naming to `createOrganizationsWithAllCombinationOfRole`